trab1.py

```diff
@@ -216,15 +216,17 @@ def rank(matrix):
     '''
     
     submatrices, n = squares(matrix)
-    
+    print(n)
     #If the determinant of at least one of the sub-matrices is non-zero, we already found the rank.
     #In other words:  if det != 0 (False) /--> rank = n
     
     is_det_zero = False
     
     for submatrix in submatrices:
-        is_det_zero = is_det_zero | indirect_det(submatrix)
-        
+        is_det_zero = is_det_zero or indirect_det(submatrix)
```

Comment:
Se 1 caso de indirect_det for falso, todo mundo tem q ser falso. Portanto o operador tem q ser AND e **is_det_zero = True**